### ISSUE TITLE
Lists with trailing li characters trigger MD032

### ISSUE THREAD

**Issue by mattdaviscodes** [DavidAnson/markdownlint#529]
### Summary

Lists with one or more valid elements and followed by a trailing hyphen or period trigger error MD032/blanks-around-lists.

### Example

Both of the lists below show the warning.

```md
# Examples

## Hyphens

- Example
-

## Period

. Example
.

```

### Expected

These lines aren't blank, so I don't think the warning should show. I'm almost always in the process of adding another list element when this warning shows.

If we're concerned about protecting against lists that end with empty list elements, I'd propose a new rule to cover that scenario specifically.

**Comment by DavidAnson** [comment]
Period as used in your second example is not a list item marker, but I see the behavior you describe in the first example and agree that warning does not seem appropriate.

**Comment by mattdaviscodes** [comment]
Thank you!
